lang/test/org/partiql/lang/syntax/SqlParserDateTimeTests.kt

```diff
@@ -25,6 +34,567 @@ class SqlParserDateTimeTests : SqlParserTestBase() {
                 project = projectList(projectExpr(date(2021, 3, 10))),
                 from = scan(id("foo"))
             )
+        },
+        DateTimeTestCase("TIME '02:30:59'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME (3) '12:59:31'") {
+            time(12, 59, 31, 0, 3, null)
+        },
+        DateTimeTestCase("TIME '23:59:59.9999'") {
+            time(23, 59, 59, 999900000, 9, null)
+        },
+        DateTimeTestCase("TIME (7) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 7, null)
+        },
+        DateTimeTestCase("TIME (9) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 9, null)
+        },
+        DateTimeTestCase("TIME (0) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 0, null)
+        },
+        DateTimeTestCase("TIME (10) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 9, null)
+        },
```

Comment:
Since the precision is beyond the maximum value of 9, this should be an error case, no?

Reply:
Not really. I checked with the Postgres and it does allow specifying the precision beyond the limit. In that case the value is rounded to the default maximum precision. (9 in this case).

Reply:
Ok, that's odd. I guess it just issues a warning.